### PR TITLE
Fix translations installation dir

### PR DESCRIPTION
**Pull request by tim77**
This is continuation of #614

### CODE REVIEW

**Comment by tim77** [comment]
Now it works. Sorry for being in a hurry. But please review it.

**Comment by Shatur** [comment]
Application name should be application name, not the ID.

**Comment by tim77** [comment]
Dunno how to fix this. I wanted to keep canonical `QLibraryInfo::location(QLibraryInfo::TranslationsPath))` in `appsettings.cpp` and in same time fix issue with space in path (`Crow Translate`). `PROJECT_NAME` also OK, but APPID is used everywhere now as Freedesktop standard to avoid conflicts.

**Comment by Shatur** [comment]
> but APPID is used everywhere now as Freedesktop standard to avoid conflicts.

Sure. But I don't think that application ID should also be the application name. Qt looks for translations in `org-name/app-name/translations` folder, yes. Is it cause issues with Fedora?
I looking at other Qt projects, like https://github.com/AntiMicroX/antimicroX and looks like the ID is `io.github.antimicrox.antimicrox`. But translation folder is `/usr/share/antimicrox/translations/`.

**Comment by tim77** [comment]
Maybe `PROJECT_NAME` would be OK? [Example](https://github.com/JonMagon/KDiskMark/blob/master/src/main.cpp#L10C57-L10C69).

**Comment by Shatur** [comment]
Let's go with `PROJECT_NAME`. Could you apply it in other non-display places, like https://github.com/crow-translate/crow-translate/blob/cfb1ca4b453d93785aa04ccfda5054f8e53af389/CMakeLists.txt#L20 ?

**Comment by tim77** [comment]
Done. Not sure for `MACOSX_BUNDLE_BUNDLE_NAME` should we change this for Mac or not.

File: CMakeLists.txt [hunk 1]
```diff
@@ -30,7 +30,7 @@ set(CPACK_RPM_FILE_NAME RPM-DEFAULT)
 set(CPACK_RPM_PACKAGE_LICENSE GPLv3)
 set(CPACK_RPM_PACKAGE_GROUP Applications/Text)
 set(CPACK_RPM_COMPRESSION_TYPE lzma)
-set(CPACK_NSIS_DISPLAY_NAME ${APPLICATION_NAME})
+set(CPACK_NSIS_DISPLAY_NAME ${PROJECT_NAME})
```
Comment (Shatur) [inline CMakeLists.txt:33]: I would like to keep display names as `Crow Translate`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -15,9 +15,9 @@ if(UNIX)
     set(METAINFO_FILE ${APPLICATION_ID}.metainfo.xml)
 endif()
 
-set(CPACK_PACKAGE_VENDOR ${APPLICATION_NAME})
-set(CPACK_PACKAGE_EXECUTABLES ${EXECUTABLE_NAME} ${APPLICATION_NAME})
-set(CPACK_PACKAGE_INSTALL_DIRECTORY ${APPLICATION_NAME})
+set(CPACK_PACKAGE_VENDOR ${PROJECT_NAME})
+set(CPACK_PACKAGE_EXECUTABLES ${EXECUTABLE_NAME} ${PROJECT_NAME})
```
Comment (Shatur) [inline CMakeLists.txt:19]: This one too.

File: CMakeLists.txt [hunk 3]
```diff
@@ -15,9 +15,9 @@ if(UNIX)
     set(METAINFO_FILE ${APPLICATION_ID}.metainfo.xml)
 endif()
 
-set(CPACK_PACKAGE_VENDOR ${APPLICATION_NAME})
-set(CPACK_PACKAGE_EXECUTABLES ${EXECUTABLE_NAME} ${APPLICATION_NAME})
-set(CPACK_PACKAGE_INSTALL_DIRECTORY ${APPLICATION_NAME})
+set(CPACK_PACKAGE_VENDOR ${PROJECT_NAME})
```
Comment (Shatur) [inline CMakeLists.txt:18]: And this one.

File: CMakeLists.txt [hunk 4]
```diff
@@ -271,7 +271,7 @@ if(UNIX AND NOT APPLE)
 elseif(APPLE)
     set_target_properties(${PROJECT_NAME} PROPERTIES
         MACOSX_BUNDLE ON
-        MACOSX_BUNDLE_BUNDLE_NAME ${APPLICATION_NAME}
+        MACOSX_BUNDLE_BUNDLE_NAME ${PROJECT_NAME}
```
Comment (Shatur) [inline CMakeLists.txt:274]: Should be also `APPLICATION_NAME` I think.

**Review by Shatur — CHANGES_REQUESTED** [review]

**Review by Shatur — APPROVED** [review]
Thank you!

**Comment by Shatur** [comment]
This may affect dbus name. Could you check if it's changed?

**Comment by tim77** [comment]
It will be necessary to mention in the changelog (for Linux) that this change will unfortunately break the user’s old configuration. I hope this will be a worthwhile sacrifice and may help others avoid problems with this space character ` `.

**Comment by tim77** [comment]
> This may affect dbus name. Could you check if it's changed?

Let me test...

**Comment by Shatur** [comment]
I think it will also affect Windows users as well. But yes, I think it worth it.

**Comment by Shatur** [comment]
Looks like CI fails: https://github.com/crow-translate/crow-translate/actions/runs/6516830369/job/17700740732
Could you submit a PR to QOnlineTranslator that removes this lint? Just add it to `-cppcoreguidelines-avoid-do-while` to `.clang_tidy`?

**Comment by tim77** [comment]
Will do. As for dbus, i think we are okay:
```
dbus-send --print-reply --dest=org.freedesktop.DBus  /org/freedesktop/DBus org.freedesktop.DBus.ListNames | rg crow
      string "io.crow_translate.CrowTranslate"
```

**Comment by Shatur** [comment]
Doing a force merge since the change is correct. I will fix CI after QOnlineTranslator update.

**Pull request merged**